cmd/haplotypeGenerator/haplotypeGenerator.go

```diff
@@ -6,83 +6,83 @@ package main
+		overlappingVariants := make([]vcf.Vcf, 0)
+		variantChan, header := vcf.GoReadToChan(s.VcfFile)
+		samplesNames := vcf.SampleNamesInOrder(header)
+		if s.Verbose > 0 {
+			fmt.Printf("working on region index %v.\n", i)
+		}
+		for currVariant := range variantChan {
+			if interval.Overlap(currVariant, regions[i]) && vcf.IsSubstitution(currVariant) {
+				overlappingVariants = append(overlappingVariants, currVariant)
+			}
+		}
 
-		overlappingVariants := interval.Query(tree, regions[i], "any")
 		currIndex := genomeMap[regions[i].Chrom]
 		refHaplotype := fasta.Extract(genome[currIndex], regions[i].ChromStart, regions[i].ChromEnd, regions[i].Chrom)
-
-		outputFilename := fmt.Sprintf("%s/%s.%v.%v.fa", outdir, regions[i].Chrom, regions[i].ChromStart, regions[i].ChromEnd)
-
-		// fmt.Printf("output filename: %s\n", outputFilename)
-
+		outputFilename := fmt.Sprintf("%s/%s.%v.%v.fa", s.OutDir, regions[i].Chrom, regions[i].ChromStart, regions[i].ChromEnd)
 		out := fileio.EasyCreate(outputFilename)
-		sampleHaplotypes := make([]fasta.Fasta, len(header.Samples)*2)
 
-		// fmt.Printf("last line of header: \n%s", header.Text[len(header.Text)-1])
-
-		// fmt.Printf("length of sample haplotypes: %v.\n", len(sampleHaplotypes))
-		// fmt.Printf("length of header.Samples: %v.\n", len(header.Samples))
+		if s.Verbose > 0 {
+			fmt.Printf("output filename: %s\n", outputFilename)
+			fmt.Printf("last line of header: \n%s\n", header.Text[len(header.Text)-1])
+			fmt.Printf("length of header.Samples: %v.\n", len(header.Samples))
+		}
 
-		for j := range sampleHaplotypes {
-			sampleHaplotypes[j] = fasta.Copy(refHaplotype)
-			currSampleIndex := j / 2
-			if j%2 == 0 {
-				sampleHaplotypes[j].Name = fmt.Sprintf("%s_A", samplesNames[currSampleIndex])
-			} else {
-				sampleHaplotypes[j].Name = fmt.Sprintf("%s_B", samplesNames[currSampleIndex])
+		for j := range samplesNames {
+			// first we calculate the haplotype for A, and write it out
+			currHaplotype = fasta.Copy(refHaplotype)
+			currHaplotype.Name = fmt.Sprintf("%s_A", samplesNames[j])
+			if s.Verbose > 0 {
+				fmt.Printf("working on haplotype: \n%s", currHaplotype.Name)
 			}
-
-			// fmt.Printf("working on haplotype: \n%s", sampleHaplotypes[j].Name)
-
 			for k := range overlappingVariants {
-				if j%2 == 0 {
-					currState = int(overlappingVariants[k].(vcf.Vcf).Samples[currSampleIndex].Alleles[0])
-					if currState > 0 {
-						sampleHaplotypes[j].Seq[overlappingVariants[k].(vcf.Vcf).Pos-regions[i].ChromStart-1] = dna.StringToBase(overlappingVariants[k].(vcf.Vcf).Alt[currState-1])
-					}
-				} else {
-					currState = int(overlappingVariants[k].(vcf.Vcf).Samples[currSampleIndex].Alleles[1])
-					if currState > 0 {
-						sampleHaplotypes[j].Seq[overlappingVariants[k].(vcf.Vcf).Pos-regions[i].ChromStart-1] = dna.StringToBase(overlappingVariants[k].(vcf.Vcf).Alt[currState-1])
-					}
+				currState = int(overlappingVariants[k].Samples[j].Alleles[0])
```

Comment:
I am a little confused how you know what SNPs are on haplotype A vs B. Is that something specified by the VCF file? With alleles[0] ?

Reply:
Yes, this is specified on the VCF